operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml

```diff
@@ -0,0 +1,194 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: kubeapiservers.operator.openshift.io
+spec:
+  group: operator.openshift.io
+  names:
+    kind: KubeAPIServer
+    plural: kubeapiservers
+    singular: kubeapiserver
+  scope: Cluster
+  preserveUnknownField: false
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      description: KubeAPIServer provides information to configure an operator to
+        manage kube-apiserver.
+      type: object
+      required:
+      - spec
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          type: object
+          properties:
+            failedRevisionLimit:
+              description: failedRevisionLimit is the number of failed static pod
+                installer revisions to keep on disk and in the api -1 = unlimited,
+                0 or unset = 5 (default)
+              type: integer
+              format: int32
+            forceRedeploymentReason:
+              description: forceRedeploymentReason can be used to force the redeployment
+                of the operand by providing a unique string. This provides a mechanism
+                to kick a previously failed deployment and provide a reason why you
+                think it will work this time instead of failing again on the same
+                config.
+              type: string
+            logLevel:
+              description: logLevel is an intent based logging for an overall component.  It
+                does not give fine grained control, but it is a simple way to manage
+                coarse grained logging choices that operators have to interpret for
+                their operands.
+              type: string
+            managementState:
+              description: managementState indicates whether and how the operator
+                should manage the component
+              type: string
+              pattern: ^(Managed|Unmanaged|Force|Removed)$
```

Comment:
this is wrong

Reply:
Last commit adds patches where they should exist, iow. here. The question remains whether kcm-o and ks-o should have a validation similar to kas-o, iow. only `Managed|Force` allowed or like the default generated. @sttts did some tests and nothing happened when setting kcm-o state to `Removed`, @deads2k thoughts?

Reply:
kcm and ks should be able to recover when set to Removed, and then to Managed by the user. So we should survive that. With ka this is different why we added the extra validaiton.

Reply:
Without a kcm and without a ks, how would the operator pods (from deployments) that would re-install these static pods be created and scheduled?  Removed needs to be ignored to ensure that the system remains running.